src/current/v24.3/ldap-authentication.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Configure LDAP Authentication
+summary: Learn how to configure CockroachDB to use LDAP for user authentication with directory services like Active Directory.
+toc: true
+---
+
+CockroachDB supports authentication and authorization using LDAP (Lightweight Directory Access Protocol) with services such as Active Directory. This allows you to integrate your existing directory service with CockroachDB for centralized user management and access control.
+
+This page describes how to configure CockroachDB user authentication using LDAP. You can also configure user authorization (role-based access control) with LDAP, to set users' CockroachDB roles based on group memberships in the directory, but this is not required.
+
+## Overview
+
+LDAP authentication in CockroachDB uses a 'search and bind' approach:
+
+1. CockroachDB first connects to the LDAP server using a service account (bind DN).
+2. It searches for the authenticating user using configurable search criteria.
+3. If found, it attempts to bind to LDAP using the user's credentials.
+4. Upon successful binding, the user is authenticated to CockroachDB.
+
+This method supports both standard LDAP servers and Active Directory, with secure LDAPS connections over TLS.
+
+{{site.data.alerts.callout_info}}
+LDAP authentication cannot be used for the `root` user or other admin users. These credentials must be managed separately using password authentication to ensure continuous administrative access, especially if the LDAP service becomes unavailable.
```

Comment:
```suggestion
LDAP authentication is not supported for the `root` user or other admin users. These credentials must be managed separately using password authentication to ensure continuous administrative access if the LDAP server or service becomes unavailable.
```
What do we mean by "admin users"? Users with the admin role, or an identity literally called "admin"?

Reply:
This is the same as the rule for GSSAPI provided here: https://www.cockroachlabs.com/docs/v21.2/gssapi_authentication#configuring-the-cockroachdb-node
refer to pt.7
```
Setting the server.host_based_authentication.configuration [cluster setting](https://www.cockroachlabs.com/docs/v21.2/cluster-settings) to this particular value makes it mandatory for all non-root users to authenticate using GSSAPI. The root user is always an exception and remains able to authenticate using a valid client cert or a user password.
```
The rule is enforced by unstated unchangeable root access rule mentioned here: https://www.cockroachlabs.com/docs/stable/security-reference/authentication#the-unstated-unchangeable-root-access-rule

> LDAP authentication is not supported for the `root` user or other admin users.
This needs a slight change. We cannot use root user or other reserved roles for LDAP authN. 
Reserved roles can be a node user or have prefix `pg_` or `crdb_internal_`.  Node users are not supposed to be authenticating via cluster SSO/db console. `pg_` and `crdb_internal_` are special roles I think which are used for running cluster jobs or some specific operations are not created directly(need to verify this).

Reply:
@souravcrl [Unless you object] I'm just going to say that LDAP cannot be used to authenticate the root user or other users with the admin role. I don't think we need to mention `node`, `pg_*` and `crdb_internal*` unless questions arise.

Side note: is there a similar setting here as in GSSAPI that *only* allows users (with the above root/admin exceptions) to auth via LDAP; no fallback? (If I'm reading GSSAPI correctly.) Or is that something we might add later?

Reply:
> I'm just going to say that LDAP cannot be used to authenticate the root user or other users with the admin role.

We can just say
LDAP cannot be used to authenticate the root user or other reserved identities.

> is there a similar setting here as in GSSAPI that only allows users (with the above root/admin exceptions) to auth via LDAP; no fallback?

There is no such setting currently for GSSAPI. Only the HBA conf mandates a rule for root user to have cert-password auth which supersedes all other HBA entries. For admin user there is no such requirement. Hence GSSAPI only enforces root user cannot be used. Admin users can still use GSSAPI.

Admin users can use LDAP also. But we don't need to mention it as it is not enforced/mandated anywhere.